### ISSUE TITLE
coq_makefile fails if there is a "\r" in _CoqProject on Linux

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#16183]
Trying exactly what's in the utilities chapter in the doc In 8.15.2.  On Ubuntu.  I believe it was working recently.

```
$ coq_makefile -f _CoqProject -o CoqMakefile
Unknown option
Usage summary:

coq_makefile .... [file.v] ... [file.ml[ig]?] ... [file.ml{lib,pack}]
  ... [-I dir] ... [-R physicalpath logicalpath]
  ... [-Q physicalpath logicalpath] ... [VARIABLE = value]
  ...  [-arg opt] ... [-docroot path] [-f file] [-o file]
  [-h] [--help]

Full list of options:

[file.v]: Coq file to be compiled
[file.ml[ig]?]: Objective Caml file to be compiled
[file.ml{lib,pack}]: ocamlbuild-style file that describes a Objective Caml
  library/module
[-I dir]: look for Objective Caml dependencies in "dir"
[-R physicalpath logicalpath]: look for Coq dependencies recursively
  starting from "physicalpath". The logical path associated to the
  physical path is "logicalpath".
[-Q physicalpath logicalpath]: look for Coq dependencies starting from
  "physicalpath". The logical path associated to the physical path
  is "logicalpath".
[VARIABLE = value]: Add the variable definition "VARIABLE=value"
[-arg opt]: send option "opt" to coqc
[-docroot path]: Install the documentation in this folder, relative to
  "user-contrib".
[-f file]: take the contents of file as arguments
[-o file]: output should go in file file (recommended)
        Output file outside the current directory is forbidden.
[-h]: print this usage summary
[--help]: equivalent to [-h]

$ more _CoqProject
-R . Jimtest

testcd.v

**Comment by SkySkimmer** [comment]
Sounds like it's hitting a non printing character (whatever it outputs after `Unknown option `), or you have an empty command line argument.

**Comment by ppedrot** [comment]
IIRC AltGr+Space (or maybe Shift+Space?) on the default configuration of Ubuntu produces a non-breaking space, which is probably what you're observing here.

**Comment by SkySkimmer** [comment]
Fixed by https://github.com/coq/coq/pull/15757

**Comment by jfehrle** [comment]
It looks like I was running the 8.15 coq_makefile though the bug fix is in 8.16.  Unfortunately, you can't run coq_makefile from a dune build tree.

**Comment by SkySkimmer** [comment]
> Unfortunately, you can't run coq_makefile from a dune build tree.

It works fine for me
May need `dune build coq-core.install` so the template makefile is in _build/install
